### ISSUE TITLE
Error when running BlobDetector on a selection of an ImageStack

### ISSUE THREAD

**Issue by JennyVoPhamhi** [spacetx/starfish#1490]
#### Description
Error thrown when trying to `run` a `BlobDetector` on a selection of an `ImageStack`.

#### Steps/Code to Reproduce
```
use_test_data = os.getenv("USE_TEST_DATA") is not None
experiment = data.ISS(use_test_data=use_test_data)
fov = experiment.fov()
imgs = fov.get_image(FieldOfView.PRIMARY_IMAGES)

masking_radius = 15
filt = Filter.WhiteTophat(masking_radius, is_volume=False)
filtered_imgs = filt.run(imgs, verbose=True, in_place=False)

sel={Axes.ROUND: 0, Axes.CH: 0, Axes.ZPLANE: 0}
single_plane_filtered = filtered_imgs.sel(sel)

p = DetectSpots.BlobDetector(
    min_sigma=1,
    max_sigma=10,
    num_sigma=10,
    threshold=0.01,
    measurement_type='mean',
)

intensities2 = p.run(single_plane_filtered)
```
#### Expected Results
Return an `IntensityTable`.

#### Actual Results
```
---------------------------------------------------------------------------
RemoteTraceback                           Traceback (most recent call last)
RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/Users/jenny.vo-phamhi/Library/Python/3.6/lib/python/site-packages/starfish/core/imagestack/imagestack.py", line 862, in _processing_workflow
    data=unshaped_numpy_array.reshape(shape),
ValueError: cannot reshape array of size 23218560 into shape (1,1,1,1044,1390)
"""

The above exception was the direct cause of the following exception:

ValueError                                Traceback (most recent call last)
<ipython-input-13-6dd71b503a32> in <module>
----> 1 intensities2 = p.run(single_plane_filtered)

~/Library/Python/3.6/lib/python/site-packages/starfish/core/pipeline/algorithmbase.py in helper(*args, **kwargs)
     44         @functools.wraps(func)
     45         def helper(*args, **kwargs):
---> 46             result = func(*args, **kwargs)
     47             # Scenario 1, Filtering, ApplyTransform
     48             if isinstance(result, ImageStack):

~/Library/Python/3.6/lib/python/site-packages/starfish/core/spots/_detect_spots/blob.py in run(self, primary_image, blobs_image, blobs_axes, n_processes, *args)
    157             measurement_function=self.measurement_function,
    158             n_processes=n_processes,
--> 159             radius_is_gyration=False)
    160 
    161         return intensity_table

~/Library/Python/3.6/lib/python/site-packages/starfish/core/spots/_detect_spots/detect.py in detect_spots(data_stack, spot_finding_method, spot_finding_kwargs, reference_image, reference_image_max_projection_axes, measurement_function, radius_is_gyration, n_processes)
    239             func=spot_finding_method,
    240             group_by={Axes.ROUND, Axes.CH},
--> 241             n_processes=n_processes
    242         )
    243         intensity_table = concatenate_spot_attributes_to_intensities(spot_attributes_list)

~/Library/Python/3.6/lib/python/site-packages/starfish/core/imagestack/imagestack.py in transform(self, func, group_by, verbose, n_processes, **kwargs)
    844             # Note: results is [None, ...] if executing an in-place workflow
    845             # Note: this return must be inside the context manager or the Pool will deadlock
--> 846             return list(zip(results, selectors))
    847 
    848     @staticmethod

/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/multiprocessing/pool.py in next(self, timeout)
    733         if success:
    734             return value
--> 735         raise value
    736 
    737     __next__ = next                    # XXX

ValueError: cannot reshape array of size 23218560 into shape (1,1,1,1044,1390)
```

#### Versions
Darwin-17.5.0-x86_64-i386-64bit
Python 3.6.5 (default, Apr 25 2018, 14:23:58) 
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.1)]
NumPy 1.17.0
SciPy 1.3.1
scikit-image 0.15.0
pandas 0.25.0
sklearn 0.21.3
xarray 0.12.3
sympy 1.4
starfish 0.1.4

**Comment by shanaxel42** [comment]
Hey @JennyVoPhamhi! so this is caused by #1268 and is high on our priorities to fix. The issue has to do with our backed memory buffer not reshaping after a selection operation happens. To get you unblocked you can use this total hack to recreate the memory buffer `new_stack = ImageStack.from_numpy(single_plane_filtered.xarray.values) and then  p.run(new_stack)` it should resolve the issue until we can get this fixed properly!

**Comment by kevinyamauchi** [comment]
Thanks, @shanaxel42!

**Comment by JennyVoPhamhi** [comment]
That works, thanks @shanaxel42!

**Comment by shanaxel42** [comment]
Going to close this particular issue since it will be fixed with #1268
